### ISSUE TITLE
found markup element with unexpected name 

### ISSUE THREAD

**Issue by pre22** [dotnet/core#5052]
Problem encountered on https://dotnet.microsoft.com/learn/aspnet/blazor-tutorial/run
Operating System: windows

Provide details about the problem you're experiencing. Include your operating system version, exact error message, code sample, and anything else that is relevant.

**Comment by carlossanlop** [comment]
> Provide details about the problem you're experiencing. Include your operating system version, exact error message, code sample, and anything else that is relevant.

**Comment by endepointe** [comment]
![deliss](https://user-images.githubusercontent.com/30739491/127308482-1cfca6cd-48f1-4948-a6ee-f5d62a2da78e.png)
![deliss2](https://user-images.githubusercontent.com/30739491/127308490-d4077dd4-1d60-4639-bf01-d15d02fc59cf.png)
I am using windows 10, os build 19042.1110, Windows Feature Experience Pack 120.2212.3530.0

Visual Studio Code has the most up to date version yet warnings still appear with the default components provided by the blazorwasm template tool.

It does not have an effect on the application, it is just ugly. I could probably turn these warnings off somewhere but I do not want to disable warnings that I may need.

**Comment by danroth27** [comment]
@endepointe What version of .NET are you using and what version of the C# extension do you have installed?

**Comment by endepointe** [comment]
C# Extension version: 1.23.13
Host .NET version: 5.0.8

**Comment by danroth27** [comment]
@endepointe If you update the C# extension to version 1.23.14 does it resolve the issue?

**Comment by endepointe** [comment]
Unfortunately, it did not.

![uddel](https://user-images.githubusercontent.com/30739491/127445164-046c0bcf-166c-4634-ac5f-37a950558bbf.png)

**Comment by danroth27** [comment]
@endepointe Thanks for checking that! To diagnose what's going on, please use the "Razor: Report an issue" command in VS Code and follow the instructions to collect the relevant diagnostic info and create a new issue so that we can investigate.

**Comment by endepointe** [comment]
Hi, I am resuming my project on my days off today but I don't see the "Razor: Report an issue" command in VS Code. I do not get a lot of time to really work on my project and thus do not want to spend time doing this kind of debugging in vs code. It zaps all energy and motivation for things I actually care about. 

If you could just get me the command or whatever it is that you need, that would be great.

**Comment by danroth27** [comment]
@endepointe If you have the [C# VS Code extension](https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.csharp) installed, then the "Razor: Report an issue" command should be available from the [VS Code Command Palette](https://code.visualstudio.com/docs/getstarted/userinterface#_command-palette) when you search for "Razor". You'll also need to turn on verbose logging for Razor, but the command should walk you through that.

**Comment by endepointe** [comment]
The third party components are working yet there are still warning underlines. I deleted the default <NavLink/> components that contained the same warning lines.

[Client - 5:35:36 AM] -- Starting Issue Data Collection-- 
OmniSharp.Extensions.LanguageServer.Shared.LspRequestRouter: Finding descriptors for textDocument/hover | Method='textDocument/hover'
OmniSharp.Extensions.LanguageServer.Server.Matchers.TextDocumentMatcher: Found attributes 1, razor::file:///c:/dev/EP_CompanyManagement/Client/Shared/Navbar.razor | Count='1' Attributes='razor::file:///c:/dev/EP_CompanyManagement/Client/Shared/Navbar.razor'
OmniSharp.Extensions.LanguageServer.Server.Matchers.TextDocumentMatcher: Looking for handler for descriptors textDocument/hover | Method='textDocument/hover'
OmniSharp.Extensions.LanguageServer.Server.Matchers.TextDocumentMatcher: Checking handler textDocument/hover:Microsoft.AspNetCore.Razor.LanguageServer.Hover.RazorHoverEndpoint | Method='textDocument/hover' Handler='Microsoft.AspNetCore.Razor.LanguageServer.Hover.RazorHoverEndpoint'
OmniSharp.Extensions.LanguageServer.Server.Matchers.TextDocumentMatcher: Registration options OmniSharp.Extensions.LanguageServer.Protocol.Models.HoverRegistrationOptions | OptionsName='OmniSharp.Extensions.LanguageServer.Protocol.Models.HoverRegistrationOptions'
OmniSharp.Extensions.LanguageServer.Server.Matchers.TextDocumentMatcher: Document Selector [**/*.{cshtml,razor}] | DocumentSelector='[**/*.{cshtml,razor}]'
OmniSharp.Extensions.LanguageServer.Server.Matchers.TextDocumentMatcher: Handler Selected: Microsoft.AspNetCore.Razor.LanguageServer.Hover.RazorHoverEndpoint  via [**/*.{cshtml,razor}] (targeting OmniSharp.Extensions.JsonRpc.IJsonRpcRequestHandler`2[[OmniSharp.Extensions.LanguageServer.Protocol.Models.HoverParams, OmniSharp.Extensions.LanguageProtocol, Version=0.18.0.0, Culture=neutral, PublicKeyToken=6d868dff454e6022],[OmniSharp.Extensions.LanguageServer.Protocol.Models.Hover, OmniSharp.Extensions.LanguageProtocol, Version=0.18.0.0, Culture=neutral, PublicKeyToken=6d868dff454e6022]]) | Handler='Microsoft.AspNetCore.Razor.LanguageServer.Hover.RazorHoverEndpoint' Id='' DocumentSelector='[**/*.{cshtml,razor}]' HandlerInterface='OmniSharp.Extensions.JsonRpc.IJsonRpcRequestHandler`2[[OmniSharp.Extensions.LanguageServer.Protocol.Models.HoverParams, OmniSharp.Extensions.LanguageProtocol, Version=0.18.0.0, Culture=neutral, PublicKeyToken=6d868dff454e6022],[OmniSharp.Extensions.LanguageServer.Protocol.Models.Hover, OmniSharp.Extensions.LanguageProtocol, Version=0.18.0.0, Culture=neutral, PublicKeyToken=6d868dff454e6022]]'
OmniSharp.Extensions.JsonRpc.ProcessScheduler: Queueing Parallel:textDocument/hover:120 request for processing | Type='Parallel' Name='textDocument/hover:120'
OmniSharp.Extensions.JsonRpc.InputHandler: Starting: Processing request textDocument/hover 120 | Method='textDocument/hover' ResponseId='120'
OmniSharp.Extensions.LanguageServer.Shared.LspRequestRouter: Starting: Routing Request (120) textDocument/hover | Id='120' Method='textDocument/hover'
OmniSharp.Extensions.LanguageServer.Shared.LspRequestRouter: Converting params for Request (120) textDocument/hover to OmniSharp.Extensions.LanguageServer.Protocol.Models.HoverParams | Id='120' Method='textDocument/hover' Type='OmniSharp.Extensions.LanguageServer.Protocol.Models.HoverParams'
OmniSharp.Extensions.LanguageServer.Shared.LspRequestRouter: Converting params for Notification textDocument/hover to OmniSharp.Extensions.LanguageServer.Protocol.Models.HoverParams | Method='textDocument/hover' Type='OmniSharp.Extensions.LanguageServer.Protocol.Models.HoverParams'
OmniSharp.Extensions.LanguageServer.Shared.LspRequestRouter: Finding descriptors for razor/languageQuery | Method='razor/languageQuery'
Microsoft.AspNetCore.Razor.LanguageServer.Hover.RazorHoverEndpoint: Found hover info items. | 
OmniSharp.Extensions.JsonRpc.ProcessScheduler: Queueing Parallel:razor/languageQuery:121 request for processing | Type='Parallel' Name='razor/languageQuery:121'
OmniSharp.Extensions.JsonRpc.InputHandler: Starting: Processing request razor/languageQuery 121 | Method='razor/languageQuery' ResponseId='121'
OmniSharp.Extensions.LanguageServer.Shared.LspRequestRouter: Starting: Routing Request (121) razor/languageQuery | Id='121' Method='razor/languageQuery'
OmniSharp.Extensions.LanguageServer.Shared.LspRequestRouter: Converting params for Request (121) razor/languageQuery to Microsoft.AspNetCore.Razor.LanguageServer.RazorLanguageQueryParams | Id='121' Method='razor/languageQuery' Type='Microsoft.AspNetCore.Razor.LanguageServer.RazorLanguageQueryParams'
OmniSharp.Extensions.LanguageServer.Shared.LspRequestRouter: Converting params for Notification razor/languageQuery to Microsoft.AspNetCore.Razor.LanguageServer.RazorLanguageQueryParams | Method='razor/languageQuery' Type='Microsoft.AspNetCore.Razor.LanguageServer.RazorLanguageQueryParams'
OmniSharp.Extensions.LanguageServer.Shared.LspRequestRouter: Response value was (null) | Type='(null)'
OmniSharp.Extensions.LanguageServer.Shared.LspRequestRouter: Finished: Routing Request (120) textDocument/hover in 0ms | Id='120' Method='textDocument/hover' ElapsedMilliseconds='0'
OmniSharp.Extensions.JsonRpc.InputHandler: Finished: Processing request textDocument/hover 120 in 0ms | Method='textDocument/hover' ResponseId='120' ElapsedMilliseconds='0'
Microsoft.AspNetCore.Razor.LanguageServer.RazorLanguageEndpoint: Language query request for (31, 32) = Html at (31, 32) | 
OmniSharp.Extensions.LanguageServer.Shared.LspRequestRouter: Response value was Microsoft.AspNetCore.Razor.LanguageServer.RazorLanguageQueryResponse | Type='Microsoft.AspNetCore.Razor.LanguageServer.RazorLanguageQueryResponse'
OmniSharp.Extensions.LanguageServer.Shared.LspRequestRouter: Finished: Routing Request (121) razor/languageQuery in 0ms | Id='121' Method='razor/languageQuery' ElapsedMilliseconds='0'
OmniSharp.Extensions.JsonRpc.InputHandler: Finished: Processing request razor/languageQuery 121 in 0ms | Method='razor/languageQuery' ResponseId='121' ElapsedMilliseconds='0'
[Client - 5:35:43 AM] 22 - Synchronizing 'c:\dev\EP_CompanyManagement\Client\Shared\Navbar.razor__virtual.html':
    Currently at 25, synchronizing to version '25'.
    Current host document version: '25'
    Current projected document version: '25'
[Client - 5:35:43 AM] 22 - Projected document in sync with host document
[Client - 5:35:43 AM] 22 - Projected text document in sync with data type
[Client - 5:35:43 AM] 22 - Synchronization successful!
OmniSharp.Extensions.LanguageServer.Shared.LspRequestRouter: Finding descriptors for textDocument/codeAction | Method='textDocument/codeAction'
OmniSharp.Extensions.LanguageServer.Server.Matchers.TextDocumentMatcher: Found attributes 1, razor::file:///c:/dev/EP_CompanyManagement/Client/Shared/Navbar.razor | Count='1' Attributes='razor::file:///c:/dev/EP_CompanyManagement/Client/Shared/Navbar.razor'
OmniSharp.Extensions.LanguageServer.Server.Matchers.TextDocumentMatcher: Looking for handler for descriptors textDocument/codeAction | Method='textDocument/codeAction'
OmniSharp.Extensions.LanguageServer.Server.Matchers.TextDocumentMatcher: Checking handler textDocument/codeAction:Microsoft.AspNetCore.Razor.LanguageServer.CodeActions.CodeActionEndpoint | Method='textDocument/codeAction' Handler='Microsoft.AspNetCore.Razor.LanguageServer.CodeActions.CodeActionEndpoint'
OmniSharp.Extensions.LanguageServer.Server.Matchers.TextDocumentMatcher: Registration options OmniSharp.Extensions.LanguageServer.Protocol.Models.CodeActionRegistrationOptions | OptionsName='OmniSharp.Extensions.LanguageServer.Protocol.Models.CodeActionRegistrationOptions'
OmniSharp.Extensions.LanguageServer.Server.Matchers.TextDocumentMatcher: Document Selector [**/*.{cshtml,razor}] | DocumentSelector='[**/*.{cshtml,razor}]'
OmniSharp.Extensions.LanguageServer.Server.Matchers.TextDocumentMatcher: Handler Selected: Microsoft.AspNetCore.Razor.LanguageServer.CodeActions.CodeActionEndpoint  via [**/*.{cshtml,razor}] (targeting OmniSharp.Extensions.JsonRpc.IJsonRpcRequestHandler`2[[Microsoft.AspNetCore.Razor.LanguageServer.CodeActions.Models.RazorCodeActionParams, Microsoft.AspNetCore.Razor.LanguageServer, Version=6.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60],[OmniSharp.Extensions.LanguageServer.Protocol.Models.CommandOrCodeActionContainer, OmniSharp.Extensions.LanguageProtocol, Version=0.18.0.0, Culture=neutral, PublicKeyToken=6d868dff454e6022]]) | Handler='Microsoft.AspNetCore.Razor.LanguageServer.CodeActions.CodeActionEndpoint' Id='' DocumentSelector='[**/*.{cshtml,razor}]' HandlerInterface='OmniSharp.Extensions.JsonRpc.IJsonRpcRequestHandler`2[[Microsoft.AspNetCore.Razor.LanguageServer.CodeActions.Models.RazorCodeActionParams, Microsoft.AspNetCore.Razor.LanguageServer, Version=6.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60],[OmniSharp.Extensions.LanguageServer.Protocol.Models.CommandOrCodeActionContainer, OmniSharp.Extensions.LanguageProtocol, Version=0.18.0.0, Culture=neutral, PublicKeyToken=6d868dff454e6022]]'
OmniSharp.Extensions.JsonRpc.ProcessScheduler: Queueing Parallel:textDocument/codeAction:122 request for processing | Type='Parallel' Name='textDocument/codeAction:122'
OmniSharp.Extensions.JsonRpc.InputHandler: Starting: Processing request textDocument/codeAction 122 | Method='textDocument/codeAction' ResponseId='122'
OmniSharp.Extensions.LanguageServer.Shared.LspRequestRouter: Starting: Routing Request (122) textDocument/codeAction | Id='122' Method='textDocument/codeAction'
OmniSharp.Extensions.LanguageServer.Shared.LspRequestRouter: Converting params for Request (122) textDocument/codeAction to Microsoft.AspNetCore.Razor.LanguageServer.CodeActions.Models.RazorCodeActionParams | Id='122' Method='textDocument/codeAction' Type='Microsoft.AspNetCore.Razor.LanguageServer.CodeActions.Models.RazorCodeActionParams'
OmniSharp.Extensions.LanguageServer.Shared.LspRequestRouter: Converting params for Notification textDocument/codeAction to Microsoft.AspNetCore.Razor.LanguageServer.CodeActions.Models.RazorCodeActionParams | Method='textDocument/codeAction' Type='Microsoft.AspNetCore.Razor.LanguageServer.CodeActions.Models.RazorCodeActionParams'
OmniSharp.Extensions.LanguageServer.Shared.LspRequestRouter: Response value was (null) | Type='(null)'
OmniSharp.Extensions.LanguageServer.Shared.LspRequestRouter: Finished: Routing Request (122) textDocument/codeAction in 0ms | Id='122' Method='textDocument/codeAction' ElapsedMilliseconds='0'
OmniSharp.Extensions.JsonRpc.InputHandler: Finished: Processing request textDocument/codeAction 122 in 1ms | Method='textDocument/codeAction' ResponseId='122' ElapsedMilliseconds='1'
OmniSharp.Extensions.LanguageServer.Shared.LspRequestRouter: Finding descriptors for textDocument/codeAction | Method='textDocument/codeAction'
OmniSharp.Extensions.LanguageServer.Server.Matchers.TextDocumentMatcher: Found attributes 1, razor::file:///c:/dev/EP_CompanyManagement/Client/Shared/Navbar.razor | Count='1' Attributes='razor::file:///c:/dev/EP_CompanyManagement/Client/Shared/Navbar.razor'
OmniSharp.Extensions.LanguageServer.Server.Matchers.TextDocumentMatcher: Looking for handler for descriptors textDocument/codeAction | Method='textDocument/codeAction'
OmniSharp.Extensions.LanguageServer.Server.Matchers.TextDocumentMatcher: Checking handler textDocument/codeAction:Microsoft.AspNetCore.Razor.LanguageServer.CodeActions.CodeActionEndpoint | Method='textDocument/codeAction' Handler='Microsoft.AspNetCore.Razor.LanguageServer.CodeActions.CodeActionEndpoint'
OmniSharp.Extensions.LanguageServer.Server.Matchers.TextDocumentMatcher: Registration options OmniSharp.Extensions.LanguageServer.Protocol.Models.CodeActionRegistrationOptions | OptionsName='OmniSharp.Extensions.LanguageServer.Protocol.Models.CodeActionRegistrationOptions'
OmniSharp.Extensions.LanguageServer.Server.Matchers.TextDocumentMatcher: Document Selector [**/*.{cshtml,razor}] | DocumentSelector='[**/*.{cshtml,razor}]'
OmniSharp.Extensions.LanguageServer.Server.Matchers.TextDocumentMatcher: Handler Selected: Microsoft.AspNetCore.Razor.LanguageServer.CodeActions.CodeActionEndpoint  via [**/*.{cshtml,razor}] (targeting OmniSharp.Extensions.JsonRpc.IJsonRpcRequestHandler`2[[Microsoft.AspNetCore.Razor.LanguageServer.CodeActions.Models.RazorCodeActionParams, Microsoft.AspNetCore.Razor.LanguageServer, Version=6.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60],[OmniSharp.Extensions.LanguageServer.Protocol.Models.CommandOrCodeActionContainer, OmniSharp.Extensions.LanguageProtocol, Version=0.18.0.0, Culture=neutral, PublicKeyToken=6d868dff454e6022]]) | Handler='Microsoft.AspNetCore.Razor.LanguageServer.CodeActions.CodeActionEndpoint' Id='' DocumentSelector='[**/*.{cshtml,razor}]' HandlerInterface='OmniSharp.Extensions.JsonRpc.IJsonRpcRequestHandler`2[[Microsoft.AspNetCore.Razor.LanguageServer.CodeActions.Models.RazorCodeActionParams, Microsoft.AspNetCore.Razor.LanguageServer, Version=6.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60],[OmniSharp.Extensions.LanguageServer.Protocol.Models.CommandOrCodeActionContainer, OmniSharp.Extensions.LanguageProtocol, Version=0.18.0.0, Culture=neutral, PublicKeyToken=6d868dff454e6022]]'
OmniSharp.Extensions.JsonRpc.ProcessScheduler: Queueing Parallel:textDocument/codeAction:123 request for processing | Type='Parallel' Name='textDocument/codeAction:123'
OmniSharp.Extensions.JsonRpc.InputHandler: Starting: Processing request textDocument/codeAction 123 | Method='textDocument/codeAction' ResponseId='123'
OmniSharp.Extensions.LanguageServer.Shared.LspRequestRouter: Starting: Routing Request (123) textDocument/codeAction | Id='123' Method='textDocument/codeAction'
OmniSharp.Extensions.LanguageServer.Shared.LspRequestRouter: Converting params for Request (123) textDocument/codeAction to Microsoft.AspNetCore.Razor.LanguageServer.CodeActions.Models.RazorCodeActionParams | Id='123' Method='textDocument/codeAction' Type='Microsoft.AspNetCore.Razor.LanguageServer.CodeActions.Models.RazorCodeActionParams'
OmniSharp.Extensions.LanguageServer.Shared.LspRequestRouter: Converting params for Notification textDocument/codeAction to Microsoft.AspNetCore.Razor.LanguageServer.CodeActions.Models.RazorCodeActionParams | Method='textDocument/codeAction' Type='Microsoft.AspNetCore.Razor.LanguageServer.CodeActions.Models.RazorCodeActionParams'
OmniSharp.Extensions.LanguageServer.Server.Pipelines.ResolveCommandPipeline: Updating Resolve items with wrapped data for textDocument/codeAction:Microsoft.AspNetCore.Razor.LanguageServer.CodeActions.CodeActionEndpoint | Method='textDocument/codeAction' Handler='Microsoft.AspNetCore.Razor.LanguageServer.CodeActions.CodeActionEndpoint'
OmniSharp.Extensions.LanguageServer.Shared.LspRequestRouter: Response value was OmniSharp.Extensions.LanguageServer.Protocol.Models.CommandOrCodeActionContainer | Type='OmniSharp.Extensions.LanguageServer.Protocol.Models.CommandOrCodeActionContainer'
OmniSharp.Extensions.LanguageServer.Shared.LspRequestRouter: Finished: Routing Request (123) textDocument/codeAction in 0ms | Id='123' Method='textDocument/codeAction' ElapsedMilliseconds='0'
OmniSharp.Extensions.JsonRpc.InputHandler: Finished: Processing request textDocument/codeAction 123 in 0ms | Method='textDocument/codeAction' ResponseId='123' ElapsedMilliseconds='0'
[Client - 5:35:49 AM] -- Stopping Issue Data Collection--
